scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py

```diff
@@ -77,26 +77,34 @@ def validate_detector_ids(self, text):
         self._view.warning_popup("Invalid detector list.")
         return False
 
-    def validate_periods(self, text):
-        period_list = run_utils.run_string_to_list(text)
+    def validate_periods(self, period_text):
+        try:
+            period_list = run_utils.run_string_to_list(period_text)
+        except IndexError:
+            self._view.warning_popup("Entered period range in invalid format."
+                                     " Valid format is a comma seperated list of numbers or dash seperated ranges e.g. 1,3-5")
+            # An IndexError thrown here implies that the input string is not a valid
+            # number list.
+            return False
         invalid_runs = []
         current_runs = self._model._context.current_runs
 
         for run in current_runs:
             for period in period_list:
                 if period < 1:
                     invalid_runs.append(run)
+                    break
```

Comment:
is t here a way we can do this without a break?

Reply:
There probably is. What is the issue with breaking out of the loop though?

Reply:
I just think its a bit hacky, no other reason

Reply:
I don' think the use of break here is hacky. For each run we are checking if it contains all the periods requested and marking it as invalid if any period is not present. Therefore if one period is found to be invalid we don't need to evaluate the rest.

Reply:
Although could replace with something like,
```
if not all([abs(period)<num_periods(run) for period in periods]):
    invalid_runs.append(run)
```

Reply:
Have now replaced with
```
for run in current_runs:
            if any([period < 1 or self._model._context.num_periods(run) < period for period in period_list]):
                invalid_runs.append(run)
```